### PR TITLE
Suppress SSL warnings if I know what I'm doing

### PR DESCRIPTION
**Pull request by jaymzh**
I don't like that we warn on something someone clearly did on purpose (changed the default), but if we're going to do that, at least give the option to turn off the warning.

### CODE REVIEW

File: spec/unit/config_spec.rb [hunk 1]
```diff
@@ -246,6 +246,10 @@ def to_platform(*args)
           expect(Chef::Config[:ssl_verify_mode]).to eq(:verify_peer)
         end
 
+        it "Chef::Config[:suppress_ssl_warnings] defaults to false" do
+          expect(Chef::Config[:ssl_verify_mode]).to eq(false)
```
Comment (stevendanna) [inline spec/unit/config_spec.rb]: I think you mean :suppress_ssl_warnings here :)

**Comment by jaymzh** [comment]
CC @opscode/client-core

**Comment by danielsdeleo** [comment]
For the sake of argument, why not just connect to your server over HTTP?

On the other hand, those warnings were added (in the Chef 11 times) just to try to get people to change their SSL setting in advance (and therefore get them to uncover any bugs with, e.g., the trusted certs stuff), knowing that we planned to change the default in Chef 12, so I wouldn't disagree with removing them entirely. In other words, we used to have a bad default that we couldn't change without breaking compat, but now we have a safe default, so like you said, we can just presume you meant it when you changed the default and not nag you about it.

**Comment by jaymzh** [comment]
Because there's a difference in trying to protect against passive monitoring vs MITM. That's like saying "you left your window unlocked, so you might as well prop your front door open!" Not a good argument. Different features of SSL protect against different things. Further, using self-signed certs may provide some users a path to using SSL more fully.

Anyway, yes I'm happy to nuke the warning entirely instead if you'd prefer that... shall I do that?

**Comment by jaymzh** [comment]
@danielsdeleo would you prefer I make that change? I actually like it better, but it seemed more controversial.

**Comment by danielsdeleo** [comment]
I agree with your reasoning and would prefer that, but I'm hoping we get more buy in from others. @opscode/client-core ?

**Comment by jaymzh** [comment]
OK, I just did that. I can always revert that commit if people don't like it.

**Comment by lamont-granquist** [comment]
works for me

**Comment by danielsdeleo** [comment]
:+1: from me.

**Comment by stevendanna** [comment]
:+1:

**Comment by jonlives** [comment]
:+1: from me too

**Pull request merged**